bioptim/optimization/receding_horizon_optimization.py

```diff
@@ -156,16 +165,26 @@ def _advance_windows(self, sol: Solution, steps: int = 0):
                 np.ndarray(sol.states["all"].shape), interpolation=InterpolationType.EACH_FRAME
             )
             self.nlp[0].x_init.check_and_adjust_dimensions(self.nlp[0].states.shape, self.nlp[0].ns)
+
+        if self.nlp[0].u_init.type != InterpolationType.EACH_FRAME:
+            self.nlp[0].u_init = InitialGuess(
+                np.ndarray(sol.controls["all"].shape), interpolation=InterpolationType.EACH_FRAME
+            )
+            self.nlp[0].u_init.check_and_adjust_dimensions(self.nlp[0].controls.shape, self.nlp[0].ns)
+
         self.nlp[0].x_init.init[:, :] = np.concatenate(
             (sol.states["all"][:, 1:], sol.states["all"][:, -1][:, np.newaxis]), axis=1
         )
 
-    def _define_time(
-        self,
-        phase_time: Union[int, float, list, tuple],
-        objective_functions,
-        constraints,
-    ):
+        self.nlp[0].u_init.init[:, :] = np.concatenate(
+            (sol.controls["all"][:, 1:], sol.controls["all"][:, -2:-1][:, np.newaxis]), axis=1
+        )
+
+    def export_current_window(self, sol, frame):
+        self.states.append(sol.states["all"][:, frame])
+        self.controls.append(sol.controls["all"][:, frame])
```

Comment:
That does not make much sense. If I call this twice, I'll add something to the MHE class, while the name (export) suggest that I just retrieve some data. 
I would have expected something like:
`return sol.states["all"][:, frame], sol.controls["all"][:, frame]`

Reply:
This is now what is done